code/game/objects/items/grenades/ghettobomb.dm

```diff
@@ -159,3 +139,80 @@
 	update_mob()
 	explosion(src.loc,-1,-1,2, flame_range = 4)	// small explosion, plus a very large fireball.
 	qdel(src)
+
+
+// Molotov Cocktail
+/obj/item/reagent_containers/food/drinks/bottle/molotov
+	name = "molotov cocktail"
+	desc = "A empty bottle with a rag in it. Needs to be filled with a flammable liquid before being lit on fire and chucked at someone you don't like."
+	icon = 'icons/fallout/objects/guns/explosives.dmi'
+	lefthand_file = 'icons/fallout/onmob/weapons/special_lefthand.dmi'
+	righthand_file = 'icons/fallout/onmob/weapons/special_righthand.dmi'
+	icon_state = "molotov"
+	item_state = "ied"
+	list_reagents = list()
+	var/list/accelerants = list(	/datum/reagent/consumable/ethanol, /datum/reagent/fuel, /datum/reagent/clf3, /datum/reagent/phlogiston,
+							/datum/reagent/napalm, /datum/reagent/hellwater, /datum/reagent/toxin/plasma, /datum/reagent/toxin/spore_burning)
+	var/active = 0
+
+/obj/item/reagent_containers/food/drinks/bottle/molotov/CheckParts(list/parts_list)
+	..()
+	var/obj/item/reagent_containers/food/drinks/bottle/B = locate() in contents
+	if(B)
+		icon_state = B.icon_state
+		B.reagents.copy_to(src,100)
+		if(!B.isGlass)
+			desc += " You're not sure if making this out of a carton was the brightest idea."
+			isGlass = FALSE
```

Comment:
```suggestion
	var/obj/item/reagent_containers/food/drinks/bottle/bottle_found = locate() in contents
	if(!bottle_found)
		return
	icon_state = bottle_found.icon_state
	bottle_found.reagents.copy_to(src, 100)
	if(!bottle_found.isGlass)
		desc = "[desc ] You're not sure if making this out of a carton was the brightest idea."
		isGlass = FALSE
```
* Make use of early returns to diminish horizontal drift.
* Make use of good informative variables, instead of single-letter upper case ones.